src/client/pingclient-rs/src/buffer.rs

```diff
@@ -0,0 +1,327 @@
+// Copyright 2019 Twitter, Inc.
```

Comment:
This whole file is copy-pasted from the pingserver - we should factor out into a separate buffer library - maybe in rustcommon?